pandas/core/indexes/base.py

```diff
@@ -5847,7 +5834,7 @@ def trim_front(strings: List[str]) -> List[str]:
     Trims zeros and decimal points.
     """
     trimmed = strings
-    while len(strings) > 0 and all(x[0] == " " for x in trimmed):
+    while trimmed and all(x.startswith(" ") for x in trimmed):
```

Comment:
this looks like it changes the logic by checking trimmed instead of strings?

perf difference between startswith vs x[0] == " "?

Reply:
True, but AFAIKT this condition is only needed if `trimmed` is an empty list (because `all([])` will always be `True`), so if we do an early return then we can remove it.

Regarding perf differences:
```
In [15]: %timeit 'foobarfdsfsdfsdafsdafhdlsafhgsdlafhsdlafhsda'.startswith('f')
110 ns ± 3.31 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)

In [16]: %timeit 'foobarfdsfsdfsdafsdafhdlsafhgsdlafhsdlafhsda'[0] == 'f'
24.1 ns ± 1.05 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)
```
so I'll go back to `[0] == ' '` (cc @ivanovmg )